### PR TITLE
Remove ClippedCamera3D

### PR DESCRIPTION
**Pull request by reduz**
* Usage was always confusing for users
* The ability to generate a pyramid shape was moved to Camera3D
* SpringArm3D now casts using the camera pyramid shape if no shape is supplied.

### CODE REVIEW

**Comment by QbieShay** [comment]
Note that this makes impossible to use just the raycast shape with the camera. I personally don't find it an issue.

**Comment by reduz** [comment]
@QbieShay Yeah, you can no longer use raycast with the camera, but I think it makes sense as now the camera always projects the correct pyramidal volume, hence it is impossible it clips.

**Comment by QbieShay** [comment]
Yes, I don't think raycasting with camera is enough of a usecase for us to worry about it. Good work!

**Comment by Calinou** [comment]
This can be done in a future PR, but the scene conversion code should probably convert existing ClippedCamera (`3.x`) and ClippedCamera3D (early `master`) nodes to Camera3D. This should also be done with InterpolatedCamera.

File: scene/3d/spring_arm_3d.cpp [hunk 1]
```diff
@@ -133,17 +134,32 @@ void SpringArm3D::process_spring() {
...
 	if (shape.is_null()) {
-		motion = Vector3(cast_direction * (spring_length));
-		PhysicsDirectSpaceState3D::RayResult r;
-		bool intersected = get_world_3d()->get_direct_space_state()->intersect_ray(get_global_transform().origin, get_global_transform().origin + motion, r, excluded_objects, mask);
-		if (intersected) {
-			real_t dist = get_global_transform().origin.distance_to(r.position);
-			dist -= margin;
-			motion_delta = dist / (spring_length);
+		Camera3D *camera = nullptr;
+		for (int i = get_child_count() - 1; 0 <= i; --i) {
```
Comment (aaronfranke) [inline scene/3d/spring_arm_3d.cpp:141]: What is the reason for interating backwards over cameras? If it's necessary, I think `i >= 0` is better than `0 <= i`.

File: scene/3d/spring_arm_3d.cpp [hunk 2]
```diff
@@ -133,17 +134,32 @@ void SpringArm3D::process_spring() {
...
+			camera = Object::cast_to<Camera3D>(get_child(i));
+			if (camera) {
+				break;
+			}
+		}
+
+		if (camera != nullptr) {
+			//use camera rotation, but spring arm position
```
Comment (aaronfranke) [inline scene/3d/spring_arm_3d.cpp:149]:
```suggestion
			// Use camera rotation, but spring arm position.
```

File: doc/classes/Camera3D.xml [hunk 3]
```diff
@@ -42,6 +42,11 @@
 				Returns the camera's frustum planes in world space units as an array of [Plane]s in the following order: near, far, left, top, right, bottom. Not to be confused with [member frustum_offset].
 			</description>
 		</method>
+		<method name="get_pyramid_shape_rid">
+			<return type="RID" />
+			<description>
+			</description>
```
Comment (aaronfranke) [inline doc/classes/Camera3D.xml:48]:
```suggestion
			<description>
				Returns the RID of a pyramid shape encompassing the camera's view frustum, ignoring the camera's near plane. The tip of the pyramid represents the position of the camera.
			</description>
```

**Review by aaronfranke — COMMENTED** [review]
Makes a lot more sense this way, and works as expected.

File: scene/3d/spring_arm_3d.cpp (hunk 1 above)
Comment (reduz) [reply scene/3d/spring_arm_3d.cpp:141]: No idea, the code was like this before, so I left it as-is.

Comment (aaronfranke) [reply scene/3d/spring_arm_3d.cpp:141]: Was it? I can't find `0 <= i` in any of the removals in this PR.

Comment (akien-mga) [reply scene/3d/spring_arm_3d.cpp:141]: It's copied from just below:
```
scene/3d/spring_arm_3d.cpp
154:    for (int i = get_child_count() - 1; 0 <= i; --i) {
```

**Review by aaronfranke — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**